src/vs/workbench/contrib/webview/electron-browser/webviewService.ts

```diff
@@ -48,6 +48,6 @@ export class ElectronWebviewService implements IWebviewService {
 	}
 
 	setIcons(id: string, iconPath: WebviewIcons | undefined): void {
-		this._iconManager.setIcons(id, iconPath);
+		this._configService.getValue('workbench.iconTheme') ? this._iconManager.setIcons(id, iconPath) : this._iconManager.setIcons(id, undefined);
```

Comment:
The icon service itself should be aware of when the icon setting changes. You can use the `IConfigurationService.onDidChangeConfiguration` event to be notified of this